### PR TITLE
1006 submit creation  launch modal with 

### PR DESCRIPTION
**Pull request by hdinia**

### CODE REVIEW

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx [hunk 1]
```diff
@@ -51,54 +60,57 @@ function CreateVariantModal(props: Props) {
...
-        </>
-      }
+      titleIcon={AddCircleIcon}
+      open={open}
+      onCancel={onClose}
+      onSubmit={handleSubmit}
+      config={{
+        defaultValues: {
+          name: "",
+          sourceId: "",
```
Comment (pl-buiquang) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx]: set the default sourceId to parentId (well you should add back the parentId props then)

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/index.tsx [hunk 2]
```diff
@@ -136,7 +136,6 @@ function InformationView(props: Props) {
           open={openVariantModal}
           onClose={() => setOpenVariantModal(false)}
           tree={tree}
-          parentId={study.id}
```
Comment (pl-buiquang) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/index.tsx:141]: readd that to have a default value

File: webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx [hunk 3]
```diff
@@ -15,153 +19,118 @@ interface PropType {
...
+    setToggle(!isToggled);
+  };
 
-  const handleChange = (key: string, value: string | number) => {
-    setCandidate({ ...candidate, [key]: value });
+  const handleSubmit = (data: SubmitHandlerPlus<XpansionCandidate>) => {
+    const { values } = data;
+
+    if (values) {
+      if (isToggled) delete values["max-investment"];
```
Comment (pl-buiquang) [inline webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: put clear accolades.
Also delete max-unit and unit-size when isToggled is false

File: webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx [hunk 4]
```diff
@@ -15,153 +19,118 @@ interface PropType {
...
+        <>
+          <Fieldset fullFieldWidth>
+            <StringFE label={t("global.name")} name="name" control={control} />
+
+            <SelectFE
+              label={t("xpansion.link")}
+              options={links.map((link) => `${link.area1} - ${link.area2}`)}
+              name="link"
+              control={control}
+              fullWidth
```
Comment (pl-buiquang) [inline webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: this one is required

File: webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx [hunk 5]
```diff
@@ -15,153 +19,118 @@ interface PropType {
...
+  ////////////////////////////////////////////////////////////////
+
+  const handleToggle = () => {
+    setToggle(!isToggled);
+  };
 
-  const handleChange = (key: string, value: string | number) => {
-    setCandidate({ ...candidate, [key]: value });
+  const handleSubmit = (data: SubmitHandlerPlus<XpansionCandidate>) => {
+    const { values } = data;
```
Comment (pl-buiquang) [inline webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: concat with default values (or there will be missing values)

**Review by pl-buiquang — CHANGES_REQUESTED** [review]

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx [hunk 6]
```diff
@@ -1,48 +1,57 @@
...
   const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
-  const [studyName, setStudyName] = useState<string>("");
-  const [versionSourceList, setVersionSourceList] = useState<
-    Array<GenericInfo>
-  >([]);
-  const [versionSource, setVersionSource] = useState<string>(parentId);
 
-  const onSave = async () => {
-    if (!studyName) {
+  ////////////////////////////////////////////////////////////////
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx]: No block for hooks

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx [hunk 7]
```diff
@@ -1,48 +1,57 @@
...
+  // State
+  ////////////////////////////////////////////////////////////////
+
+  const [sourceList, setSourceList] = useState<Array<GenericInfo>>([]);
+
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleSubmit = async (data: SubmitHandlerPlus) => {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx]: Add type argument to `SubmitHandlerPlus`

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx [hunk 8]
```diff
@@ -1,48 +1,57 @@
...
+    const { sourceId, name } = data.values;
+
+    if (!name) {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx]: `if` not necessary

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx [hunk 9]
```diff
@@ -1,48 +1,57 @@
...
         variant: "error",
       });
       return;
     }
     try {
-      const newId = await createVariant(versionSource, studyName);
-      setStudyName("");
-      onClose();
-      navigate(`/studies/${newId}`);
+      if (sourceId && name) {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx]: Check of `name` not necessary

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx [hunk 10]
```diff
@@ -51,54 +60,57 @@ function CreateVariantModal(props: Props) {
     }
   };
 
+  ////////////////////////////////////////////////////////////////
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx:62]: Hook on top without block

File: webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx [hunk 11]
```diff
@@ -51,54 +60,57 @@ function CreateVariantModal(props: Props) {
...
+  // Hooks
+  ////////////////////////////////////////////////////////////////
+
   useEffect(() => {
-    setVersionSourceList(createListFromTree(tree));
+    setSourceList(createListFromTree(tree));
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/HomeView/InformationView/CreateVariantModal/index.tsx]: `setVersionList`

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/CreateAreaModal.tsx [hunk 12]
```diff
@@ -1,76 +1,62 @@
...
-    posY: number,
-    color: string
-  ) => {
-    if (!isStringEmpty(id)) {
-      onSave(id, posX, posY, color);
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleSubmit = (data: SubmitHandlerPlus) => {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/CreateAreaModal.tsx]: Add type arg in `SubmitHandlerPlus`

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/CreateAreaModal.tsx [hunk 13]
```diff
@@ -1,76 +1,62 @@
...
+    const { name } = data.values;
+
+    if (!isStringEmpty(name)) {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/CreateAreaModal.tsx]: Add required in field instead

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/index.tsx [hunk 14]
```diff
@@ -128,9 +128,9 @@ function Map() {
 
   const onSave = async (
     name: string,
-    posX: number,
-    posY: number,
-    color: string
+    posX = 0,
+    posY = 0,
+    color = "rgb(230, 108, 44)"
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/index.tsx:133]: Why do you change this ?

File: webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx [hunk 15]
```diff
@@ -15,153 +19,118 @@ interface PropType {
...
-    "annual-cost-per-mw": 0,
-  });
-  const [toggleView, setToggleView] = useState<boolean>(true);
-  const tabLinks = links.map((item) => `${item.area1} - ${item.area2}`);
+  const [t] = useTranslation();
+
+  ////////////////////////////////////////////////////////////////
+  // State
+  ////////////////////////////////////////////////////////////////
+  const [isToggled, setToggle] = useState<boolean>(true);
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: No block for hooks

File: webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx (hunk 5 above)
Comment (skamril) [reply webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: `data.values` include default values (!= `data.dirtyValues`).
But there is no default values here?

File: webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx [hunk 16]
```diff
@@ -15,153 +19,118 @@ interface PropType {
...
+  const handleToggle = () => {
+    setToggle(!isToggled);
+  };
 
-  const handleChange = (key: string, value: string | number) => {
-    setCandidate({ ...candidate, [key]: value });
+  const handleSubmit = (data: SubmitHandlerPlus<XpansionCandidate>) => {
+    const { values } = data;
+
+    if (values) {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: `data.values` cannot be empty

**Review by skamril — CHANGES_REQUESTED** [review]

File: webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx (hunk 5 above)
Comment (pl-buiquang) [reply webapp/src/components/App/Singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: didn't seem to work though..

**Review by skamril — APPROVED** [review]

**Review by pl-buiquang — APPROVED** [review]

**Pull request merged**